src/varname.jl

```diff
@@ -231,37 +359,76 @@ julia> @varname(x).indexing
 ()
 
 julia> @varname(x[1]).indexing
-((1,),)
+(@lens _[1])
 
 julia> @varname(x[:, 1]).indexing
-((Colon(), 1),)
+(@lens _[Colon(), 1])
 
 julia> @varname(x[:, 1][2]).indexing
-((Colon(), 1), (2,))
+(@lens _[Colon(), 1][2])
 
 julia> @varname(x[1,2][1+5][45][3]).indexing
-((1, 2), (6,), (45,), (3,))
+(@lens _[1, 2][6][45][3])
 ```
 
 !!! compat "Julia 1.5"
     Using `begin` in an indexing expression to refer to the first index requires at least
     Julia 1.5.
 """
-macro varname(expr::Union{Expr, Symbol})
-    return esc(varname(expr))
+macro varname(expr::Union{Expr, Symbol}, concretize=false)
```

Comment:
This really needs to be `concretize::Bool=false`, otherwise the `if` will fail if you pass an arbitrary expression.  (Or you splice the expression in a returned `if`, but I'd prefer the check to happen at expansion time.)

Reply:
This is exactly why I made it `false` :+1:

Reply:
Yeah but the way you wrote it, withouth the type bound, it will still allow other expressions, which are not boolean literals, and then fail with a `TypeError` somwhere inside, instead of a `MethodError` directly.

Reply:
Ah I see what you're saying, haha. I completely misparsed it earlier.